src/create-rpc-server.ts

```diff
@@ -1,45 +1,106 @@
+/* eslint-disable no-console */
+
 import { createServer } from 'node:http'
+import { decode } from 'node:querystring'
 import { writeHeapSnapshot } from 'node:v8'
+import { enable } from '@libp2p/logger'
 
 export interface RpcServerOptions {
   apiPort?: number
   apiHost?: string
 }
 
+const resources: Record<string, Record<string, Parameters<typeof createServer>[1]>> = {
+  '/api/v0/nodejs/gc': {
+    GET: (req, res) => {
+      if (globalThis.gc == null) {
+        // maybe we're running in a non-v8 engine or `--expose-gc` wasn't passed
+        res.writeHead(503, { 'Content-Type': 'text/plain' })
+        res.end('Service Unavailable')
+
+        return
+      }
+
+      // force nodejs to run garbage collection
+      globalThis.gc?.()
+      res.writeHead(200, { 'Content-Type': 'text/plain' })
+      res.end('OK')
+    }
+  },
+  '/api/v0/nodejs/heapdump': {
+    GET: (req, res) => {
+      // force nodejs to generate a heapdump
+      // you can analyze the heapdump with https://github.com/facebook/memlab#heap-analysis-and-investigation to get some really useful insights
+      const filename = writeHeapSnapshot(`./snapshot-dir/${(new Date()).toISOString()}.heapsnapshot`)
```

Comment:
Not directly related to this PR, but why do we not just stream in the response to the user instead of writing to disk?

Reply:
Makes sense, can be done in a follow up though.

Reply:
https://github.com/libp2p/js-libp2p-amino-dht-bootstrapper/pull/153